### ISSUE TITLE
Support path attributes that don't end with ﻿`.rs`

### ISSUE THREAD

**Issue by bvanjoi** [rust-lang/rust-analyzer#17089]
For example:

```rs
// main.rs
#[path = "a.suffix"]
mod a;
fn main() {}

// a.suffix
pub const A: i32 = 1;
```

`rustc main.rs` can compile successfully, but ra will report errors:

<img width="473" alt="image" src="https://github.com/rust-lang/rust-analyzer/assets/30187863/addff1c1-9e54-4d89-9fc4-ebcad4e6ce47">

**Comment by poliorcetics** [comment]
Relevant docs: https://doc.rust-lang.org/reference/items/modules.html?highlight=%23%5Bpath%20%3D#the-path-attribute and wow I didn't know all of the specifications of the `#[path]` attribute

**Comment by poliorcetics** [comment]
I think the issue is not that rust-analyzer doesn't understand such paths.

I tried it and got the error but the moment I open the `a.suffix` file and tell my editor it's a Rust file, the error disappears. I think either Rust-Analyzer or the editor filter out the files sent to RA by extension.

The fix is probably to make R-A request files in `path` attributes when it doesn't know about them (i.e, they don't match `*.rs`). I'm very much unsure about how to fix this though.

**Comment by poliorcetics** [comment]
Yep, it happens here: https://github.com/rust-lang/rust-analyzer/blob/55d9a533b309119c8acd13061581b43ae8840823/crates/rust-analyzer/src/main_loop.rs#L173-L177

**Comment by poliorcetics** [comment]
# Workaround for people ending up here

1. Open the file that R-A fails to detect
2. Tell your editor it's a Rust file

Editors that correctly implement the LSP spec should pick it up and send it to R-A. You'll have to do it each and every time you restart R-A though (quitting and opening the editor will do it for example)
